emulator/callbacks_test.go

```diff
@@ -0,0 +1,241 @@
+/*
+ * Flow Emulator
+ *
+ * Copyright Flow Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package emulator
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/onflow/cadence"
+	jsoncdc "github.com/onflow/cadence/encoding/json"
+	flowsdk "github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestParseSchedulerProcessedEvent(t *testing.T) {
+	serviceAddress := flow.HexToAddress("0x01")
+	expectedEventType := fmt.Sprintf("A.%s.UnsafeCallbackScheduler.CallbackProcessed", serviceAddress.String())
+
+	tests := []struct {
+		name           string
+		event          flowsdk.Event
+		serviceAddress flow.Address
+		expectedLimit  uint64
+		expectedID     []byte
+		expectError    bool
+		errorContains  string
+	}{
+		{
+			name: "valid event with ID=1 and effort=1000",
+			event: flowsdk.Event{
+				Type: expectedEventType,
+				Value: cadence.NewEvent([]cadence.Value{
+					cadence.NewUInt64(1),    // ID
+					cadence.NewUInt64(1000), // executionEffort
+				}).WithType(cadence.NewEventType(
+					nil,
+					"A.01.UnsafeCallbackScheduler.CallbackProcessed",
```

Comment:
Here and in the other test cases below: The type ID looks incorrect, the address part should be full length